modules/db/src/main/java/org/mule/module/db/internal/domain/connection/OracleDbConnection.java

```diff
@@ -93,6 +98,36 @@ private Method getCreateArrayOfMethod(Connection delegate)
         return createArrayMethod;
     }
 
+    @Override
+    protected void resolveLobsForArrays(String typeName, Object[] elements) throws SQLException
+    {
+        String collectionTypeName = getTypeFor(typeName);
+
+        if(collectionTypeName != null)
+        {
+            typeName = collectionTypeName;
+        }
+
+        Map<Integer, String> dataTypes = getLobFieldsDataTypeInfo(typeName);
+
+        if (dataTypes.keySet().isEmpty())
+        {
+            logger.warn("No catalog information was found for the typename {}. No lob resolution will be performed", typeName);
+        }
+
+        Integer index;
+        for (Map.Entry entry : dataTypes.entrySet())
+        {
+            index = (Integer) entry.getKey();
+            String dataTypeName = dataTypes.get(index);
```

Comment:
if you iterate through the entry set as sonar request do not use get again. Get the value from the entry.